src/Adapter/Cache/Clearer/SymfonyCacheClearer.php

```diff
@@ -78,13 +72,18 @@ public function clear()
             $input = new ArrayInput([
                 'command' => 'cache:clear',
                 '--no-optional-warmers' => true,
-                '--env' => _PS_ENV_,
+                '--env' => $kernel->getEnvironment(),
             ]);
 
             $output = new NullOutput();
             $application->run($input, $output);
 
             Hook::exec('actionClearSf2Cache');
         });
+
+        // Unlock is registered in another seaprate function to make sure it will be called no matter what
+        register_shutdown_function(function () use ($kernel) {
+            $kernel->unlocksCacheClear();
```

Comment:
Do you mean having a timeout after which the file is unlocked? There is no such feature with `flock` as far as I know. 

But anyway even if the file is not unlocked manually it will be automatically unlocked after the process that locked it ends, so if indeed someone calls `exit` in `actionClearSf2Cache` even though they SHOULD NEVER DO IT then the file will be unlocked.